SimTracker/SiPhase2Digitizer/plugins/PixelDigitizerAlgorithm.cc

```diff
@@ -260,63 +256,6 @@ bool PixelDigitizerAlgorithm::isAboveThreshold(const digitizerUtility::SimHitInf
 // -- Read Bad Channels from the Condidion DB and kill channels/module accordingly
 //
 void PixelDigitizerAlgorithm::module_killing_DB(const Phase2TrackerGeomDetUnit* pixdet) {
-  bool isbad = false;
-  uint32_t detID = pixdet->geographicalId().rawId();
-  int ncol = pixdet->specificTopology().ncolumns();
-  if (ncol < 0)
-    return;
-  std::vector<SiPixelQuality::disabledModuleType> disabledModules = siPixelBadModule_->getBadComponentList();
-
-  SiPixelQuality::disabledModuleType badmodule;
-  for (const auto& mod : disabledModules) {
-    if (detID == mod.DetID) {
-      isbad = true;
-      badmodule = mod;
-      break;
-    }
-  }
-
-  if (!isbad)
-    return;
-
-  signal_map_type& theSignal = _signal[detID];  // check validity
-  if (badmodule.errorType == 0) {               // this is a whole dead module.
-    for (auto& s : theSignal)
-      s.second.set(0.);  // reset amplitude
-  } else {               // all other module types: half-modules and single ROCs.
-    // Get Bad ROC position:
-    // follow the example of getBadRocPositions in CondFormats/SiPixelObjects/src/SiPixelQuality.cc
-    std::vector<GlobalPixel> badrocpositions;
-    for (size_t j = 0; j < static_cast<size_t>(ncol); j++) {
-      if (siPixelBadModule_->IsRocBad(detID, j)) {
-        std::vector<CablingPathToDetUnit> path = fedCablingMap_->pathToDetUnit(detID);
-        for (auto const& p : path) {
-          const PixelROC* myroc = fedCablingMap_->findItem(p);
-          if (myroc->idInDetUnit() == j) {
-            LocalPixel::RocRowCol local = {39, 25};  //corresponding to center of ROC row, col
-            GlobalPixel global = myroc->toGlobal(LocalPixel(local));
-            badrocpositions.push_back(global);
-            break;
-          }
-        }
-      }
-    }
-
-    for (auto& s : theSignal) {
-      std::pair<int, int> ip;
-      if (pixelFlag_)
-        ip = PixelDigi::channelToPixel(s.first);
-      else
-        ip = Phase2TrackerDigi::channelToPixel(s.first);
-
-      for (auto const& p : badrocpositions) {
-        for (auto& k : badPixels_) {
-          if (p.row == k.getParameter<int>("row") && ip.first == k.getParameter<int>("row") &&
-              std::abs(ip.second - p.col) < k.getParameter<int>("col")) {
-            s.second.set(0.);
-          }
-        }
-      }
-    }
-  }
+  throw cms::Exception("PixelDigitizerAlgorithm") << "Trying to kill modules from the pixel digitizer."
+                                                  << " This method is not yet implemented!";
```

Comment:
would be good to know if Phase-2 tracker experts agree with this (for the time being)

Reply:
@emiglior please comment.

Reply:
For the time being, the only use case I can think of is implementing failure scenarios for the Inner Tracker.

To be thorough, I can raise this point at tomorrow Tracker ph-2 sim meeting, but as nobody is working on this topic it is probably ok to proceed with the PR.

@suchandradutta FYI